python/python/lance/dataset.py

```diff
@@ -1988,6 +1989,17 @@ def write_dataset(
         a custom class that defines hooks to be called when each fragment is
         starting to write and finishing writing.
     """
+    try:
+        # Huggingface datasets
+        import datasets
+
+        if isinstance(data_obj, datasets.Dataset):
+            if schema is None:
+                schema = data_obj.features.arrow_schema
+            data_obj = data_obj.data.to_batches()
+    except ImportError:
+        pass
```

Comment:
if it is a HF dataset but `datasets` is not installed, then the user will likely get a very cryptic error message no?

Reply:
changed to use lazy loading